debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -446,7 +451,50 @@ protected void execute() {
                                         final Object[] row = new Object[numColumns];
                                         while (rs.next()) {
                                             for (int i = 0, j = 1; i != numColumns; ++i, ++j) {
-                                                row[i] = rs.getObject(j);
+                                                Column actualColumn = table.columns().get(i);
+                                                if(actualColumn.jdbcType() == Types.TIME) {
```

Comment:
How about extracting all this into its own method?

Reply:
Would even be an idea to write separate test for that. It'd still be an IT (as DB access is needed), but separate from all the connector logic.

Reply:
I can do that.